### ISSUE TITLE
Fix Request: AvoidUnguardedAssignmentToNonFinalFieldsInSharedObjects - static inner class of @Component

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#492]
example:

```
@Component
public final class SomeUtil {
    public static class SomeID {
        private String userID;
        public String getUserID() {
            if (userID == null) {
                userID = initialUserID(); // <- false positive
            }
            return userID;
        }
   }
}
```
